test/e2e/validation_test.go

```diff
@@ -310,7 +310,8 @@ func (tc *testContext) sshSetup() error {
 func (tc *testContext) runPowerShellSSHJob(name, command, ip string) (string, error) {
 	// Modify command to work when default shell is the newer Powershell version present on Windows Server 2022.
 	powershellDefaultCommand := command
-	if tc.CloudProvider.GetType() == config.VSpherePlatformType || tc.CloudProvider.GetType() == config.AzurePlatformType {
+	if tc.CloudProvider.GetType() == config.VSpherePlatformType ||
+		tc.CloudProvider.GetType() == config.GCPPlatformType || tc.CloudProvider.GetType() == config.AzurePlatformType {
 		powershellDefaultCommand = strings.ReplaceAll(command, "\\\"", "\"")
```

Comment:
@sebsoto the cloud provider is not the driving factor here. This modification is required due to Windows Server 2022.

Consider using the existing `tc.windowsServerVersion` flag in the condition statement for a more accurate decision. See this [commit](https://github.com/openshift/windows-machine-config-operator/pull/1091/commits/c312195b164d86f351c3a5b5f98ffacd1d795bbf#diff-19a6ecdd2ef106d645523b5e6ddc22fd6ddfd1608477cd0b0fbd85113c8c245d).

For example:
```go
	if tc.windowsServerVersion != "2019" {
		powershellDefaultCommand = strings.ReplaceAll(command, "\\\"", "\"")
	}
```

Reply:
@jrvaldes thats a good point, but I don't think that method is usable until the AWS 2022 PR is merged. If I try and do that now, the AWS 2022 jobs will fail

Reply:
What I mean is the jobs that are set for use with 2022 will fail due to the wrong powershell parsing being used for them

Reply:
The environment variable introduced in https://github.com/openshift/release/pull/29791 and is ready to use.

Reply:
Yes but aws-e2e-ccm-install and the aws upgrade job have that env var set to ""
this will cause the test suite to detect them as running server 2022 and use the wrong powershell format because of that

Reply:
I don't think so. the Windows Server 2019 is [hard-coded](https://github.com/openshift/windows-machine-config-operator/blob/f619b40ec2175f58926547a2e2b94e5b1d8a22c0/test/e2e/providers/aws/aws.go#L128).

My point is that we should stop using `tc.CloudProvider.GetType()` as criteria to tweak the PS commands. It was valid with AWS+WS2022 and is now a reality with GCP+WS2022.

Reply:
we cannot do `powershellDefaultCommand = strings.ReplaceAll(command, "\\\"", "\"")` on VMs running Windows server 2019. The AWS upgrade and ccm jobs are currently running 2019, but checking that environment variable will tell us that they are running 2022.